db/migration/V021__create_assessments.sql

```diff
@@ -0,0 +1,137 @@
+  principal_id BIGINT NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  INDEX curriculum_assessments_curriculum_id (curriculum_id),
+  FOREIGN KEY (curriculum_id) REFERENCES curriculums(id),
+  INDEX curriculum_assessments_activity_id (activity_id),
+  FOREIGN KEY (activity_id) REFERENCES activities(id),
+  INDEX curriculum_assessments_principal_id (principal_id),
+  FOREIGN KEY (principal_id) REFERENCES principals(id),
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE program_assessments (
+  id BIGINT NOT NULL AUTO_INCREMENT,
+  program_id BIGINT NOT NULL,
+  assessment_id BIGINT NOT NULL,
+  available_after TIMESTAMP NOT NULL,
+  due_date TIMESTAMP NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  INDEX program_assessments_program_id (program_id),
+  FOREIGN KEY (program_id) REFERENCES programs(id),
+  INDEX program_assessments_assessment_id (assessment_id),
+  FOREIGN KEY (assessment_id) REFERENCES curriculum_assessments(id),
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE program_participants (
+  id BIGINT NOT NULL AUTO_INCREMENT,
+  principal_id BIGINT NOT NULL,
+  program_id BIGINT NOT NULL,
+  role_id BIGINT NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  INDEX program_participants_principal_id (principal_id),
+  FOREIGN KEY (principal_id) REFERENCES principals(id),
+  INDEX program_participants_program_id (program_id),
+  FOREIGN KEY (program_id) REFERENCES programs(id),
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE assessment_submissions (
+  id BIGINT NOT NULL AUTO_INCREMENT,
+  assessment_id BIGINT NOT NULL,
+  principal_id BIGINT NOT NULL,
+  state_id BIGINT NOT NULL,
+  score INT,
+  opened_at TIMESTAMP NOT NULL,
+  submitted_at TIMESTAMP NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  INDEX assessment_submissions_assessment_id (assessment_id),
+  FOREIGN KEY (assessment_id) REFERENCES program_assessments(id),
+  INDEX assessment_submissions_principal_id (principal_id),
+  FOREIGN KEY (principal_id) REFERENCES principals(id),
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE assessment_questions (
+  id BIGINT NOT NULL AUTO_INCREMENT,
+  assessment_id BIGINT NOT NULL,
+  title TEXT NOT NULL,
+  description TEXT,
+  question_type_id BIGINT NOT NULL,
+  correct_answer_id BIGINT,
+  max_score INT,
+  sort_order INT NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  INDEX assessment_questions_assessment_id (assessment_id),
+  FOREIGN KEY (assessment_id) REFERENCES curriculum_assessments(id),
```

Comment:
This should probably be an `ON DELETE CASCADE` constraint so as not to leave orphaned questions.